torch/csrc/lazy/core/shape_inference.cpp

```diff
@@ -369,6 +371,10 @@ std::vector<Shape> compute_shape_native_layer_norm_backward(const at::Tensor& gr
   return shapes;
 }
 
+TORCH_API std::vector<Shape> compute_shape_new_empty_strided(const at::Tensor & self, at::IntArrayRef size, at::IntArrayRef stride, c10::optional<at::ScalarType> dtype, c10::optional<at::Layout> layout, c10::optional<at::Device> device, c10::optional<bool> pin_memory) {
```

Comment:
just curious, how come these 2 ops needed compute_shape functions, they were the only 2 that aren't possible to call with meta tensors for some reason?

Reply:
Yep that's a totally reasonable question. `new_empty_strided` in core has a `CompositeExplicitAutograd` kernel, but it probably should work with meta tensors.

What makes it annoying in the codegen is that we have way of statically knowing which of our composite ops do/don't have meta tensor support. You can find out by running the op until it decompose into its base ops, and hoping that all of the base operators are structured / have meta kernel implementations.

I handled all of the `view_copy` ops directly in the codegen because it happens to be the case that **all* view/view_copy ops work with meta tensors.

Reply:
One way to clean up some of these hand-written shape inference functions would be to add an allow-list of ops in the codegen that are composite, but that we happen to know work with meta tensors, and just call the meta tensor codegen infra on them.

That would probably let us remove a good number of these functions (probably future work though?). In the great end state when we're at (close to) 100% meta coverage, we can just assume that all composite ops work with meta tensors and always use the codegen for them.

Reply:
> add an allow-list of ops in the codegen that are composite, but that we happen to know work with meta tensors, and just call the meta tensor codegen infra on them.

Yea, this seems doable- especially now that we have that extra 'non-native' precedent.

> (probably future work though?)

yea, i'm good with that being 'future work'